brainatlas_api/atlas_gen/atlas_scripts/ratatlas.py

```diff
@@ -0,0 +1,364 @@
+
+volume = load_labelled_volume(load_any(str(base_data_fld / "annotated.tif")))
+
+root = extract_volume_surface(volume)
+
+write(root, str(meshes_dir_path / "0.obj"))
+
+# First create a mesh for every minor region
+volume_data = load_any(str(base_data_fld / "annotated.tif"))
+for i, region in tqdm(clean_hierarchy.iterrows()):
+    structure = [
+        s for s in structures if s["acronym"] == region["abbreviation"]
+    ][0]
+    savepath = str(
+        meshes_dir_path
+        / f'{structure["id"]}.obj'.replace("/", "-").replace("\\", "-")
+    )
+    if os.path.isfile(savepath):
+        continue
+
+    vol = np.zeros_like(volume_data)
+
+    if not np.isin(np.float(region.minor), volume_data):
+        # print(f'{region.abbreviation} doesnt seem to appear in annotated dataset')
+        continue
+
+    vol[volume_data == np.float32(region.minor)] = 1
+    if np.max(vol) < 1:
+        raise ValueError
+
+    write(extract_volume_surface(Volume(vol)), savepath)
+
+
+# Create a mesh for every submajor and major region
+for i, submajor in tqdm(submajors.iterrows()):
+    structure = [
+        s
+        for s in structures
+        if s["acronym"] == submajor["SubANC Name"].replace(" ", "-")
+    ][0]
+    savepath = str(
+        meshes_dir_path
+        / f'{structure["id"]}.obj'.replace(" ", "-")
+        .replace("/", "-")
+        .replace("\\", "-")
+    )
+    if os.path.isfile(savepath):
+        continue
+
+    regions = list(
+        clean_hierarchy.loc[
+            clean_hierarchy.submajor == submajor["SubANC Name"]
+        ].minor.values
+    )
+    if not regions:
+        continue
+
+    vol = np.zeros_like(volume_data)
+
+    for region in regions:
+        vol[volume_data == region] = 1
+
+    if np.max(vol) < 1:
+        continue
+
+    write(extract_volume_surface(Volume(vol)), savepath)
+
+
+for i, major in tqdm(majors.iterrows()):
+    if not isinstance(major["ANC Name"], str):
+        continue
+    structure = [
+        s
+        for s in structures
+        if s["acronym"] == major["ANC Name"].replace(" ", "-")
+    ][0]
+    savepath = str(
+        meshes_dir_path
+        / f'{structure["id"]}.obj'.replace(" ", "-")
+        .replace("/", "-")
+        .replace("\\", "-")
+    )
+    if os.path.isfile(savepath):
+        continue
+
+    regions = list(
+        clean_hierarchy.loc[
+            clean_hierarchy.major == major["ANC Name"]
+        ].minor.values
+    )
+    if not regions:
+        continue
+
+    vol = np.zeros_like(volume_data)
+
+    for region in regions:
+        vol[volume_data == region] = 1
+
+    if np.max(vol) < 1:
+        continue
+
+    write(extract_volume_surface(Volume(vol)), savepath)
+
+
+# ---------------------------------------------------------------------------- #
+#                            FINAL METADATA AND SAVE                           #
+# ---------------------------------------------------------------------------- #
+
+metadata_dict = {
```

Comment:
The idea of the wrap-up function that I was using in the test_atlas script is to offload this metadata generation to a place where there's also a set of checks on overall structure, types, etc.

Reply:
yeah I need to integrate your changes still, I just copied over what I had in the other repo